### ISSUE TITLE
F# Compiler <-> FSharp.Core Compatibility

### ISSUE THREAD

**Issue by ctaggart** [Microsoft/visualfsharp#555]
I put FSharp.Core 4.0.0.1 in the NuGet Gallery yesterday. I'm still a little fuzzy on compatibility of it with the compiler versions. For example, it appears that you can add it to a Visual Studio 2013 F# 3.1 project without issue. Is this correct?
https://github.com/fsharp/fsharp/issues/429#issuecomment-124573917

How independent is FSharp.Core from the Compiler? A table or list would be great.

**Comment by latkin** [comment]
If you are using the compiler from release X, it is only supported to target FSharp.Core versions that were part of release X or earlier.

It is not supported (or tested at all) to use an older-version compiler to reference a newer-version FSharp.Core.  It will probably work fine for a lot of scenarios, but liable to introduce issues in more exotic usages. I can't produce a concrete example at the moment, but will think about it.

Here's a table to help organize the different versions and releases.  I've tried to use colors to indicate the motivation for the version numbers. There are 3 axes to consider - .NET runtime version, F# language version, and portable profile.  You'll see the versioning of the portable libraries was a bit ad hoc and inconsistent in VS 2012 and 2013, but we've finally made it consistent and predictable starting with VS 2015.

![image](https://cloud.githubusercontent.com/assets/5943573/8882856/5778bcce-31ff-11e5-8157-91167eb43b10.png)

So the compiler released for a particular column above is supported to target anything from that column or to the left.
